File: practica parte 1/Ejercicio 1: Clase caja genérica/caja.py
class Caja:
    def __init__(self):
        self.contenido = None
    
    def guardar(self, item):
        self.contenido = item
    
    def obtener(self):
        return self.contenido

# Prueba
caja_num = Caja()
caja_num.guardar(100)
print("Caja numérica:", caja_num.obtener())

caja_texto = Caja()
caja_texto.guardar("Hola UMSA")
print("Caja texto:", caja_texto.obtener())

File: practica parte 1/Ejercicio 1: Clase caja genérica/src/caja.java
public class Caja<T> {
    private T contenido;
    
    public void guardar(T item) {
        this.contenido = item;
    }
    
    public T obtener() {
        return contenido;
    }

    public static void main(String[] args) {
        Caja<Integer> cajaNum = new Caja<>();
        cajaNum.guardar(100);
        System.out.println("Caja numérica: " + cajaNum.obtener());
        
        Caja<String> cajaTexto = new Caja<>();
        cajaTexto.guardar("Hola UMSA");
        System.out.println("Caja texto: " + cajaTexto.obtener());
    }
} 